internal/util/validation/validation.go

```diff
@@ -107,6 +108,52 @@ func ValidateBearerToken(token *string, path string) []error {
 	return asErrors(errs)
 }
 
+// usages for a CSR should include 'client auth'
+// TODO - do we want to add more here? or just not let the user populate this?
+func ValidateUsages(u *[]string) []error {
+	errs := field.ErrorList{}
+
+	for _, a := range *u {
+		if a == "client auth" {
+			return nil
+		}
+	}
```

Comment:
This should be two loops. 
One with fields that must be present setting an interim OK.
One with fields that must not be present bouncing it.